File: w1/greedyAlgorithms/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    int cents;
    int numCoins = 0;
    do
    {
        change = get_float("Input total change owed: ");
    }

    while (change <= 0);
    {
        cents = round(change * 100);
        int quarters = cents / 25;
        for (int i = 0; i < quarters; i++)
        {
            cents = cents - 25;
            numCoins++;
        }
        int dimes = cents / 10;
        for (int j = 0; j < dimes; j++)
        {
            cents = cents - 10;
            numCoins++;
        }
        int nickels = cents / 5;
        for (int k = 0; k < nickels; k++)
        {
            cents = cents - 5;
            numCoins++;
        }
        numCoins = numCoins + cents;
        printf("%i\n", numCoins);
    }
}

File: w6/cash.py
from cs50 import get_float

numCoins = 0

while True:
    change = get_float("Input total change owed: ")

    if change > 0:
        cents = round(change * 100)

        # Handle quarters
        quarters = int(cents / 25)
        for i in range(quarters):
            numCoins += 1
            cents = cents - 25

        # Handle quarters
        dimes = int(cents / 10)
        for j in range(dimes):
            numCoins += 1
            cents = cents - 10

        # Handle nickels
        nickels = int(cents / 5)
        for k in range(nickels):
            numCoins += 1
            cents = cents - 5

        # Handle pennies
        numCoins = numCoins + cents

        print(f"{numCoins}")

        break